web-app/js/portal/filter/ui/NumberFilterPanel.js

```diff
@@ -101,6 +101,12 @@ Portal.filter.ui.NumberFilterPanel = Ext.extend(Portal.filter.ui.BaseFilterPanel
 
         if (this.firstField.validate() && this.secondField.validate()) {
 
+            // Fixes #1696
+            if (this._operatorIsBetween() && 
```

Comment:
I think it'd be nice to extract this into a function. Something like `_shouldUpdate` then the code here can just be:
```javascript
if (!this._shouldUpdate() {
    return;
}
```

Doing that would make this code more readable (i.e. you don't need to read that boolean logic to work out what cases it is trying to catch).